config/quality/checkstyle/checkstyle.xml

```diff
@@ -1,238 +1,249 @@
+        <module name="EqualsHashCode" />
+        <module name="ExplicitInitialization" />
+        <module name="FallThrough" />
+        <module name="FinalLocalVariable" />
+        <!-- Disabled because the "this" rule already enforces that the variables will be assigned correctly -->
+        <module name="HiddenField">
+            <property name="severity" value="ignore" />
+        </module>
+        <module name="IllegalCatch" />
+        <module name="IllegalInstantiation" />
+        <module name="IllegalThrows" />
+        <module name="IllegalToken">
+            <property name="tokens" value="LITERAL_NATIVE" />
+        </module>
+        <module name="IllegalTokenText">
+            <property name="tokens" value="STRING_LITERAL" />
+            <property name="format" value="a href" />
+        </module>
+        <module name="InnerAssignment" />
+        <module name="MagicNumber" />
+        <!-- Conflicts with PMD. For experienced developers who know about hidden constructors PMD's way is more readable -->
+        <module name="MissingCtor">
+            <property name="severity" value="ignore" />
+        </module>
+        <module name="MissingSwitchDefault" />
+        <module name="ModifiedControlVariable" />
+        <module name="MultipleStringLiterals" />
+        <module name="MultipleVariableDeclarations" />
+        <module name="NestedForDepth" />
+        <module name="NestedIfDepth" />
+        <module name="NestedTryDepth" />
+        <module name="NoClone" />
+        <module name="NoFinalizer" />
+        <module name="OneStatementPerLine" />
+        <module name="OverloadMethodsDeclarationOrder" />
+        <module name="PackageDeclaration" />
+        <module name="ParameterAssignment" />
+        <module name="RequireThis" />
+        <module name="ReturnCount" />
+        <module name="SimplifyBooleanExpression" />
+        <module name="SimplifyBooleanReturn" />
+        <module name="StringLiteralEquality" />
+        <module name="SuperClone" />
+        <module name="SuperFinalize" />
+        <module name="UnnecessaryParentheses" />
+        <module name="VariableDeclarationUsageDistance" />
 
-    <module name="TreeWalker">
+        <!-- Checks for imports                              -->
+        <!-- See http://checkstyle.sourceforge.net/config_import.html -->
+        <module name="AvoidStarImport" />
+        <!-- We suppress this warning because the Java creators themselves disagree on this one, for example in the newest JUnit 5 -->
+        <module name="AvoidStaticImport">
+            <property name="severity" value="ignore" />
+        </module>
+        <module name="IllegalImport" />
+        <!-- Suppressed because we don't have so many imports that this is necessary -->
+        <!-- <module name="ImportControl"/> -->
+        <!-- Suppressed because we don't have so many imports that this is necessary -->
+        <!-- <module name="ImportOrder"/> -->
```

Comment:
Yes, I've no idea why, but the xml parser of Checkstyle crashes for a few rules because he doesn't recognize them although they are in the official documentation. If I would have ignored them anyway I just commented them out for now